### DISCUSSION TITLE
globbing and guaranteed disk ordering ?

### DISCUSSION THREAD

**Discussion by hydra3333** [trapexit/mergerfs#1350] [Q&A] [answered]
Help !  I use a new Raspberry Pi 5 with Debian bookworm.

If I have mountpoints /srv/usb3disk1 ... /srv/usb3disk8 and use glob below in fstab

`/srv/usb3disk*/mediaroot /srv/media mergerfs defaults,category.action=ff,category.create=ff,category.search=all,moveonenospc=true,dropcacheonclose=true,cache.readdir=true,cache.files=partial,lazy-umount-mountpoint=true,x-systemd.requires=/srv/usb3disk1 0 0`

is it **guaranteed** that mergerfs will mount them in **strict numerical order** ? i.e.
/srv/usb3disk1 
/srv/usb3disk2
...
/srv/usb3disk7
/srv/usb3disk8

I need that to happen since I need ffd to operate in that order :)

I tried /srv/usb3disk{1..8}/mediaroot but it treated it as text.    
I've seen fstab globbing described in other places as "depends on the underlying file system" and perhaps "no".

On reflection, I'm not sure mergerfs can affect glob to do a sort (as most people would perhaps subconsciously expect) ... can it ?    
I was doing some python stuff on the Pi and ran into that there as well where it was out of expected order, and had to sort the glob result.

While I'm asking, is there a way to "continue" lines in fstab onto the following line ? Debian bookworm.

**Comment by trapexit** [comment, marked as answer]
mergerfs uses glob function which sorts the values by default. Likely with a standard radix sort. I have no intention or reason to change the code to not sort. Naturally I can change the code to do whatever but there is no reason to do so.

I'm not aware of any way to break lines in fstab.

**Reply by hydra3333** [reply]
Thank you.  Being sorted will do just fine for my purposes (unlike the python globbing).
